sql/item_geofunc.h

```diff
@@ -1229,6 +1230,49 @@ class Item_func_sphere_distance: public Item_real_func
 };
 
 
+class Item_func_geohash: public Item_geometry_func
+{
+  double latitude;
+  double longitude;
+  uint geohash_max_output_length;
+  const double max_longitude;
+  const double min_longitude;
+  const double max_latitude;
+  const double min_latitude;
+  const uint upper_limit_output_length;
+  const std::string BASE32;
+
+  void encode_geohash(String *str);
+  void set_bit(double &max_value, double &min_value, const double &target_value,
+               std::bitset<8> &base_set, const uint &bit_index);
+public:
+  Item_func_geohash(THD *thd, Item *point, Item *max_length):
+    Item_geometry_func(thd, point, max_length),
+    max_longitude(180.0),
+    min_longitude(-180.0),
+    max_latitude(90.0),
+    min_latitude(-90.0),
+    upper_limit_output_length(100),
+    BASE32("0123456789bcdefghjkmnpqrstuvwxyz") {}
```

Comment:
Is there a reason to have these as const and set at constructor time rather than statics set at compile time? I don't see them changing anywhere.